homework/index.js

```diff
@@ -28,26 +22,139 @@
+    addRow(table, 'Fork: ', repo.forks);
+    addRow(table, 'Updated:', formatDate(repo.updated_at));
   }
 
+  function renderContributor(contributor, contributorList) {
+    const li = createAndAppend('li', contributorList);
+    const list = createAndAppend('a', li, {
+      class: 'contributor-item',
+      href: contributor.html_url,
+      target: '_blank',
+    });
+    createAndAppend('img', list, {
+      class: 'image',
+      src: contributor.avatar_url,
+    });
+    createAndAppend('span', list, {
+      class: 'contributor-name',
+      text: contributor.login,
+    });
+    createAndAppend('span', list, {
+      class: 'contribution-count',
+      text: contributor.contributions,
+    });
+    list.addEventListener('click', () => {
+      window.open(contributor.html_url, '_blank');
+    });
+    list.addEventListener('key', event => {
+      if (event.key === 'Enter') window.open(contributor.html_url, '_blank');
+    });
+  }
+  async function fetchAndRender(repo) {
+    try {
+      const contributors = await fetchJSON(repo.contributors_url);
+      const rightSide = document.getElementById('right-side');
+      rightSide.innerHTML = '';
+      createAndAppend('span', rightSide, {
+        text: 'Contributions',
+        class: 'title',
+      });
+      const contributorList = createAndAppend('ul', rightSide, {
+        class: 'contributor-list',
+      });
+      contributors.forEach(contributor =>
+        renderContributor(contributor, contributorList),
+      );
+    } catch (error) {
+      renderError(error);
+    }
+  }
+
+  function startPage(repositories) {
+    const select = document.getElementById('select');
+    repositories.sort((a, b) => a.name.localeCompare(b.name));
+    repositories.forEach((repo, index) => {
+      createAndAppend('option', select, {
+        value: index,
+        text: repo.name,
+      });
+    });
+    renderRepoDetails(repositories[select.value]);
+
+    select.addEventListener('change', () => {
+      renderRepoDetails(repositories[select.value]);
+      fetchAndRender(repositories[select.value]);
+    });
+    return repositories[select.value];
+  }
+
+  async function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    createAndAppend('p', header, {
+      class: 'header-title',
+      id: 'header-title',
+      text: 'HYF-Repositories',
+    });
+    createAndAppend('select', header, {
+      id: 'select',
+    });
+
+    const mainArea = createAndAppend('main', root, {
+      class: 'main-container',
+      id: 'main-container',
+    });
+    createAndAppend('section', mainArea, {
+      class: 'table-container',
+      id: 'left-side',
+    });
+    createAndAppend('section', mainArea, {
+      class: 'contributors-container',
+      id: 'right-side',
+    });
+
+    fetchJSON(url);
+    try {
+      const repository = await fetchJSON(url);
+      startPage(repository);
+      fetchAndRender(repository[0]);
```

Comment:
Why is this `fetchAndRender` here, and not in your `startPage` function? You already have a call to `renderRepoDetails` inside `startPage`. It makes sense to let `startPage` call both functions at start-up, as it already does so in the event handler.